src/sagemaker/jumpstart/session_utils.py

```diff
@@ -0,0 +1,203 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""This module stores SageMaker Session utilities for JumpStart models."""
+
+from __future__ import absolute_import
+
+from typing import Optional, Tuple
+from sagemaker.jumpstart.constants import DEFAULT_JUMPSTART_SAGEMAKER_SESSION
+
+from sagemaker.jumpstart.utils import get_jumpstart_model_id_version_from_resource_arn
+from sagemaker.session import Session
+from sagemaker.utils import aws_partition
+
+
+def get_model_id_version_from_endpoint(
+    endpoint_name: str,
+    inference_component_name: Optional[str] = None,
+    sagemaker_session: Session = DEFAULT_JUMPSTART_SAGEMAKER_SESSION,
+) -> Tuple[str, str, Optional[str]]:
+    """Given an endpoint name (and optionally an inference component), get the model ID and version.
```

Comment:
I suggest clarifying: "Given an endpoint name (and an inference component name if there are multiple inference components associated with the endpoint), get the model ....

Reply:
I know you are trying to fit the docstring header to a single line, but how about:
```py
"""Given an endpoint and optionally an inference component names, return the model ID and version.
"""
```